tests/integration/targets/vmware_esxi_connection/tasks/eco-vcenter.yml

```diff
@@ -0,0 +1,179 @@
+        vcenter_username: "{{ vcenter_username }}"
+        vcenter_password: "{{ vcenter_password }}"
+        vcenter_validate_certs: false
+        name: "{{ test_resource_pool }}"
+        parent: "{{ vcenter_rp_id }}"
+      register: _test_rp
+
+    - name: Gather Library Item Info
+      vmware.vmware.content_library_item_info:
+        validate_certs: false
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        library_item_name: "{{ esxi_content_library_template }}"
+        library_name: "{{ ci_resources_content_library }}"
+      register: _template
+
+    - name: Deploy Virtual ESXi Host
+      vmware.vmware_rest.vcenter_ovf_libraryitem:
+        vcenter_hostname: "{{ vcenter_hostname }}"
+        vcenter_username: "{{ vcenter_username }}"
+        vcenter_password: "{{ vcenter_password }}"
+        vcenter_validate_certs: false
+        ovf_library_item_id: "{{ (_template.library_item_info | first).id }}"
+        session_timeout: 10000
+        state: deploy
+        target:
+          resource_pool_id: "{{ _test_rp.id }}"
+        deployment_spec:
+          name: "{{ test_esxi_hostname }}"
+          accept_all_EULA: true
+          storage_provisioning: thin
+      register: _deploy
+
+    - name: Power On Host
+      vmware.vmware_rest.vcenter_vm_power:
+        state: start
+        vm: "{{ _deploy.value.resource_id.id }}"
+        vcenter_hostname: "{{ vcenter_hostname }}"
+        vcenter_username: "{{ vcenter_username }}"
+        vcenter_password: "{{ vcenter_password }}"
+        vcenter_validate_certs: false
+
+    - name: Wait For ESXi Host To Have IP
+      vmware.vmware.guest_info:
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        validate_certs: false
+        name: "{{ test_esxi_hostname }}"
+      register: _esxi_info
+      until: _esxi_info.guests[0].ipv4 is defined and _esxi_info.guests[0].ipv4
+      retries: 15
+      delay: 5
+
+    - name: Join Hosts To Cluster
+      vmware.vmware.esxi_host:
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        validate_certs: false
+        datacenter: "{{ vcenter_datacenter }}"
+        cluster: "{{ test_cluster }}"
+        esxi_host_name: "{{ _esxi_info.guests[0].ipv4 }}"
+        esxi_username: "{{ test_esxi_username }}"
+        esxi_password: "{{ test_esxi_password }}"
+        state: present
+
+    - name: Put Host In Maintenance Mode
+      vmware.vmware.esxi_maintenance_mode:
+        hostname: "{{ vcenter_hostname }}"
+        username: "{{ vcenter_username }}"
+        password: "{{ vcenter_password }}"
+        validate_certs: false
+        esxi_host_name: "{{ test_esxi_ip | default(_esxi_info.guests[0].ipv4) }}"
+        enable_maintenance_mode: true
+
+    - name: Connect Host - Idempotence
```

Comment:
I don't see when it was connected the first time. Maybe better after the `Join Hosts To Cluster` task add `Disconnect Host` task and then `Connect Host`